### ISSUE TITLE
Wayland in Docker containers

### ISSUE THREAD

**Issue by KaruroChori** [KaruroChori/vs-fltk#42]
Support for wayland was left out of the docker containers in `docker/docker-compose.yml`. 
Neither me nor @andy5995 have wayland on our systems, so any external help would be appreciated.

**Comment by andy5995** [comment]
@Jammyjamjamman  The instructions for wayland are in https://github.com/KaruroChori/vs-fltk/blob/master/docker/README.md#for-wayland-based-systems-eg-some-modern-linux-environments but haven't been tested, and once tested, the wayland service needs to be added to docker-compose.yml 

But anyone else can let use know here if they want to take this issue. :)

**Comment by andy5995** [comment]
We worked on this today, and found that if one is in Wayland, using the [x11-dev service](https://github.com/KaruroChori/vs-fltk/blob/e53443ae90163a17c830c0b0696019872188af83/docker/docker-compose.yml#L22) works, but it's not technically using Wayland.

He found some stuff about wayland in the fltk docs which mentioned setting some env vars, but trying it didn't change the result. It might be we overlooked something having worked on it for over two hours and not being fresh.

**Comment by andy5995** [comment]
Actually, using vs with Wayland in general might need a separate issue. Jammy tried in the container and on his host system. He couldn't get vs to use Wayland either way. The app worked on his host system, but didn't use Wayland.

**Comment by KaruroChori** [comment]
Please, could you disable line https://github.com/KaruroChori/vs-fltk/blob/master/meson.build#L33 ?
That is likely responsible for forcing X11.

**Comment by andy5995** [comment]
> Please, could you disable line https://github.com/KaruroChori/vs-fltk/blob/master/meson.build#L33 ? That is likely responsible for forcing X11. It is still in place as macos can only build this project with x11 but not cocoa for some extremely strange reason. #37

Disable it? Do mean the next time we test? Otherwise you could retitle and merge #37, that's the only change in that PR.

**Comment by KaruroChori** [comment]
Yes, just to see if that was the reason. In case it is what prevents wayland from working I will add some option to make sure the macos14 build does not fail.

**Comment by KaruroChori** [comment]
I just applied a patch which should remove x11 when other options are available (and forced it via options for the macos14 build). You can test from `master` directly.

**Comment by andy5995** [comment]
We did more testing today. The results didn't change, but we did find out something useful I think:

FLTK_BACKEND is ignored. When we built the libfltk separately and ran the examples from that project, it did run them under Wayland.

https://fltk.gitlab.io/fltk/wayland-devel.html

```
    if the app defines a global bool variable called fl_disable_wayland and this variable is true, the X11 leg is chosen;
    if environment variable FLTK_BACKEND is defined to string "wayland", the Wayland leg is chosen;
    if environment variable FLTK_BACKEND is defined to string "x11", the X11 leg is chosen;
    otherwise, a connection to a Wayland compositor is attempted; if it's successful, the Wayland leg is chosen; if it's not, the X11 leg is chosen.
```
